Step0 - 학습테스트 실습

brandnuws opened the pull request:
- 학습테스트 구현

P/R이 조금 늦었습니다 ㅜㅜ
사소한거라도 궁금한 포인트가 있으면 언제든 말씀 부탁드립니다 ㅎㅎ

brandnuws committed:
Add some description about 9 principles

이해가 되지 않는 부분에 대해서 조사 후 내용을 추가하였습니다.
-출처: https://velog.io/@jhp1115

brandnuws committed:
Add some description for making habit

와닿는 말이어서 눈에 띄는곳에 두려고 합니다.

brandnuws committed:
feat: stringTest 요구사항1 ~ 3 수행

brandnuws committed:
feat: setTest 요구사항1 ~ 3 수행 + stringTest 수정
-stringTest DisplayName을 조금 수정했습니다.

yomni commented on src/test/java/study/SetTest.java:
```diff
@@ -0,0 +1,54 @@
+        // given
+        // setUp()
+
+        // when
+        int actual = numbers.size();
+
+        // then
+        assertThat(actual).isEqualTo(3);
```
ㅎㅎ알고계실지 모르겠지만,
assertJ 라이브러리엔 컬렉션전용 asserttion 메서드가 별도로 존재합니다!
`.isEqualTo()` 도 좋지만, `.hasSize`로 _**이 테스트는 컬렉션의 사이즈에 대한 테스트이다**_ 라고 한 번에 보여주는 것이 어떨까요?

brandnuws replied:
> ㅎㅎ알고계실지 모르겠지만, assertJ 라이브러리엔 컬렉션전용 asserttion 메서드가 별도로 존재합니다! `.isEqualTo()` 도 좋지만, `.hasSize`로 _**이 테스트는 컬렉션의 사이즈에 대한 테스트이다**_ 라고 한 번에 보여주는 것이 어떨까요?

프로님! 리뷰해주셔서 감사해요 ㅎㅎ 아래 내용 모두 수정하고 커밋하여 resolve 처리하였습니다. 참고부탁드립니다 ㅎㅎ

yomni commented on src/test/java/study/SetTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @ParameterizedTest()
+    @CsvSource(value = {"1:true", "2:true", "3:true", "4:false", "5:false"}, delimiter = ':')
```
**가독성을 위해** multiline으로 바꿔보는 것은 어떠실까요?

```suggestion
    @CsvSource(value = {
            "1:true", 
            "2:true", 
            "3:true", 
            "4:false", 
            "5:false"}, 
        delimiter = ':')
```

yomni commented on src/test/java/study/StringTest.java:
```diff
@@ -1,13 +1,90 @@
+        String[] actual = targetString.split(",");
+
+        //then
+        assertThat(actual).contains("1");
+        assertThat(actual).contains("2");
+        assertThat(actual).containsExactly("1", "2");
```
저도 이번에 알았는데 assertThat 메소드가 **체이닝이 가능**한 것 같네요 :)
어떤게 가독성이 더 있는진 모르겠지만 일단 방법은 공유드립니다~

```suggestion
        assertThat(actual)
            .contains("1")
            .contains("2")
            .containsExactly("1", "2");
```

yomni commented on src/test/java/study/StringTest.java:
```diff
@@ -1,13 +1,90 @@
+        String[] actual = targetString.split(",");
+
+        //then
+        assertThat(actual).contains("1");
+        assertThat(actual).containsExactly("1");
```
여기서도 메소드 체이닝이 가능하겠네요

yomni commented on src/test/java/study/StringTest.java:
```diff
@@ -1,13 +1,90 @@
+    void charAt_test_expect_throwing_StringIndexOutOfBoundsException() {
+
+        assertThatThrownBy(() -> {
+            //given
+            String targetString = "abc";
```
혹시 given을 assertThatThrownBy 안에 선언하신 이유가 있으실까요~?

yomni commented on src/test/java/study/StringTest.java:
```diff
@@ -1,13 +1,90 @@
+
+            //when
+            String actual = String.valueOf(targetString.charAt(5));
+        }).isInstanceOf(IndexOutOfBoundsException.class)
+
+                //then
+                .hasMessageContaining("String index out of range: 5");
```
저 같은 경우는 when / then을 하나로 봤는데,

when : 유효하지 않은 index에 접근할 때
then : IndexOutOfBoundsException이 발생하고, 예외 문구가 `String index out of range: 5` 인지 확인한다.

로 봤습니다! 혹시 then을 hasMessageContaining으로만 분리하신 이유가 궁금합니다~!

brandnuws committed:
feat: SetTest, StringTest 코드리뷰내용 반영
-> 다중 assert시 메소드체이닝을 사용하여 간결하게 수정하였습니다.
-> when, then 부정확한 표기에 대한 수정하였습니다.
-> size를 비교하는 assert구문 수정하였습니다.
-> csvSource를 라인 구분을 통해 가독성을 높였습니다.

brandnuws committed:
feat: 문자열 계산기

brandnuws commented:
@yomni 
이게 approve되어도 후에 commit이 있으면 다시 approve받아야 merge가 열리는것 같네요 ㅋㅋㅋ... 한번 다시 approve 부탁드립니다!
